dhis-2/dhis-api/src/main/java/org/hisp/dhis/commons/util/TextUtils.java

```diff
@@ -724,4 +724,38 @@ public static String cleanUrlPathOnly(@Nonnull String baseUrl, @Nonnull String p
     String base = removeAnyTrailingSlash(baseUrl);
     return base + ("/" + path).replaceAll("/+", "/");
   }
+
+  /**
+   * Sanitizes the given input by replacing characters which are not matching any of the valid
+   * characters as specified by the given pattern with the given replacement character. Consecutive
+   * replacement characters are merged into a single character. Any leading replacement characters
+   * are removed.
+   *
+   * @param pattern the pattern of valid characters.
+   * @param input the input string.
+   * @param replacement the replacement characters for invalid characters.
+   * @return a sanitized string.
+   */
+  public static String sanitize(Pattern pattern, String input, char replacement) {
```

Comment:
Should we trim the `input`?

Reply:
Right. This was meant as a general purpose sanitize method. We can add this to the method in `ColumnNameUtils`.